06_puppy/kasunfdo/syncstore.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+func (s *SyncStore) CreatePuppy(puppy *Puppy) bool {
+	_, exists := s.Map.LoadOrStore(puppy.ID, puppy)
+	// exists is true if a record exists with given key, false if value stored.
+	return !exists
+}
+
+func (s *SyncStore) ReadPuppy(id uint64) (*Puppy, bool) {
+	value, exists := s.Map.Load(id)
+
+	if exists {
+		puppy := value.(*Puppy)
+		return puppy, true
+	}
+
+	return nil, false
+}
+
+func (s *SyncStore) UpdatePuppy(id uint64, puppy *Puppy) bool {
+	if _, exists := s.Map.Load(id); exists {
+		puppy.ID = id
```

Comment:
Same comment as on the `MapStore` implementation. Are you updating the ID correctly with respect to the map key?